src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.cpp

```diff
@@ -804,6 +804,42 @@ CHIP_ERROR GenericThreadStackManagerImpl_OpenThread<ImplClass>::_GetPollPeriod(u
     return CHIP_NO_ERROR;
 }
 
+template <class ImplClass>
+void GenericThreadStackManagerImpl_OpenThread<ImplClass>::OnDiscoveryResult(otActiveScanResult * aResult, void * aContext)
+{
+    if ((aResult == nullptr) || (aContext == nullptr))
+    {
+        return;
+    }
+
+    GenericThreadStackManagerImpl_OpenThread * threadStackInstance =
```

Comment:
Don't cast `Impl()` back to `GenericThreadStackManagerImpl_OpenThread`. Use `ImplClass*`, instead.